[RFC 0052] Update additional GenAI fields

susan-shu-c opened the pull request:
<!-- Thanks for contributing to Elastic ECS! To help us review your PR effectively and efficiently, please fill out the following sections.

Before submitting, please ensure you have:

Read the [ECS Contribution Guidelines](https://github.com/elastic/ecs/blob/main/CONTRIBUTING.md).

Signed the [Contributor License Agreement (CLA)](https://www.elastic.co/contributor-agreement) if you haven't already.

Ensured your changes align with the [ECS Guidelines and Best Practices](https://www.elastic.co/docs/reference/ecs/ecs-guidelines).

Filled in all of the sections below. -->

#### 1. What does this PR do?
<!-- Clearly describe the purpose and scope of your changes.

If this PR addresses an existing issue, please link it: Fixes #ISSUE_NUMBER or Closes #ISSUE_NUMBER. -->

* Stage 0: https://github.com/elastic/ecs/pull/2519
* Stage 1: https://github.com/elastic/ecs/pull/2525

#### 2. Which ECS fields are affected/introduced?
<!-- List all new or modified ECS fields.

For each, briefly explain its purpose and why it's needed within ECS.

If introducing new fields or making large changes to fields, please follow the [ECS RFC process](https://github.com/elastic/ecs/blob/main/rfcs/PROCESS.md)
and consider if they are generic enough for broader use cases beyond your immediate need. -->

Field | Type | Description /Usage
-- | -- | --
gen_ai.system_instructions | flattened | The system message or instructions provided to the GenAI model separately from the chat history.
gen_ai.input.messages | flattened | The chat history provided to the model as an input.
gen_ai.output.messages | flattened | Messages returned by the model where each message represents a specific model response (choice, candidate).
gen_ai.tool.definitions | flattened | The list of source system tool definitions available to the GenAI agent or model.
gen_ai.tool.call.arguments | flattened | Parameters passed to the tool call.
gen_ai.tool.call.result | flattened | The result returned by the tool call (if any and if execution was successful).

Apr 23, 2026:

This RFC also renames the existing `gen_ai.system` field introduced in RFC 0050 to `gen_ai.provider.name`, following its deprecation and replacement in OTel semconv ([ref](https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/#gen-ai-system)). The OTel mapping relation is updated from `equivalent` to `match`. The field type, level, and allowed values are unchanged.

Field | Type | Change
-- | -- | --
~~gen_ai.system~~ → gen_ai.provider.name | keyword | Renamed to align with OTel `gen_ai.provider.name`; description updated from "product" to "provider"

Changes based on OTel:
* https://github.com/open-telemetry/semantic-conventions/pull/2179/
* https://github.com/open-telemetry/semantic-conventions/pull/2702/

See changes introduced in OTel release: https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.38.0

#### 3. Why is this change necessary?
<!-- Explain the problem this PR solves or the new functionality it enables. Provide context and use cases where these changes would be beneficial. -->


#### 4. Have you added/updated documentation?
<!-- For new schema fields and larger changes, documenting the changes are part of the RFC process.

For smaller changes, it may be sufficient for documentation to be done within the schema itself, for example in the description field.

For script / generator changes, please ensure your code is documented, and consider adding separate documentation in `docs/` for larger or more complicated changes.

If you answer N/A, please explain why documentation is not needed. -->
YES / NO / N/A

#### 5. Have you built ECS and committed any newly generated files?
<!-- Build ECS with `make`, and commit any changed files in `generated/`. -->
YES / NO

#### 6. Have you run the ECS validation tests locally?
<!-- Run tests locally with `make test` and `make check`. (Optional: Briefly describe any issues encountered or resolved during validation) -->
YES / NO

#### 7. Anything else for the reviewers?
<!-- Provide any additional context, considerations, or questions for the reviewers. Are there any specific areas you'd like particular attention paid to? -->

Please see summary of feedback and concerns about flattened vs. nested [here (rfcs/text)](https://github.com/elastic/ecs/blob/79a7d1ac38893446045abdedec96747d2ecae55c/rfcs/text/0052-gen_ai-additional-fields.md#concerns)

---

#### Commit Message
<!-- This will be used as the commit message when merging the PR. Please clearly explain what change is being changed and why. -->

susan-shu-c committed:
Add proposed new fields in .yml format

susan-shu-c committed:
Add built doc files

susan-shu-c committed:
Update proposed type for tool.call.result

susan-shu-c committed:
Merge generated files from main

susan-shu-c committed:
Checkout files from main

susan-shu-c committed:
Fix typo

susan-shu-c committed:
Update generated docs

mjwolf committed:
Rebuild generated files

trisch-me commented on rfcs/text/0052/gen_ai.yaml:
```diff
@@ -0,0 +1,51 @@
+      description: The system message or instructions provided to the GenAI model separately from the chat history.
+      example: TODO
+      level: extended
+      beta: This field reuse is beta and subject to change.
```
why it’s field `reuse`?

mjwolf replied:
Sorry, I was incorrect before, the text should be `This field is beta and subject to change.` for all the beta fields here

trisch-me commented on rfcs/text/0052/gen_ai.yaml:
```diff
@@ -0,0 +1,51 @@
+    - name: system_instructions
+      type: flattened
+      description: The system message or instructions provided to the GenAI model separately from the chat history.
+      example: TODO
```
update all examples

susan-shu-c replied:
I pushed the changes, @trisch-me with all examples!

trisch-me reviewed (changes requested):
as stage 2 is a final stage, please update all examples and generate all fields

flash1293 commented:
The main difference to keep in mind is that flattened does not retain an association between the properties of an array of objects, while nested does. Taking an example from the otel repo:
```
[
  {
    "role": "system",
    "parts": [
      {
        "type": "text",
        "content": "You are a helpful bot"
      }
    ]
  },
  {
    "role": "user",
    "parts": [
      {
        "type": "text",
        "content": "Tell me a joke about OpenTelemetry"
      }
    ]
  }
]
```

With flattened, it would not be possible to query for something like `"system" role has a text like "helpful bot"`, because the data is stored like this:
```
{
  "role": ["system",  "user"],
  "parts.content": [
    "You are a helpful bot",
    "Tell me a joke about OpenTelemetry"
  ]
}
```

The association between the `role` field and the `parts.content` field is lost - in the example it looks like that it would be possible to just look at the index of the value in the array, but that doesn't work in practice for multiple reasons.

So whether it should be flattened or nested really depends on what we want to do with this data later on.

susan-shu-c committed:
Update examples

susan-shu-c committed:
Grab doc file from main

mjwolf committed:
Merge remote-tracking branch 'upstream/main' into additional-gen_ai-stage-2

susan-shu-c committed:
Update field types based on feedback tradeoffs

susan-shu-c committed:
Change field beta reuse wording

susan-shu-c committed:
Clean up wording and examples in rfcs/text

susan-shu-c committed:
Update schemas/gen_ai.yml

trisch-me committed:
Merge branch 'main' into additional-gen_ai-stage-2

Mikaayenson commented:
Migrating a slack thought to the issue for posterity. Here are a couple things we should consider before pushing this forward.
1.  The default setting for limiting nested fields on indices ( `index.mapping.nested_fields.limit` ) is 50 . If customers try to create a new index with a higher limit, they will receive the following `error: Settings [index.mapping.nested_fields.limit,index.mapping.nested_objects.limit]` are not available when running in serverless mode. It's a serverless limitation that can't be overridden without Elastic support involved. ECH its much higher. I think like 10k and can be manually overridden.
2. IINM `nested` fields are not visible in Kibana visualizations. If there are any that we would imagine want to be visualized, it would be impacted, so we may want to consider changing them to `flattened`.

susan-shu-c committed:
Comment out not-merged OTel fields

susan-shu-c committed:
Comment out not-merged OTel fields

susan-shu-c committed:
Update related OTel

susan-shu-c committed:
Remove trailing spaces via lint

susan-shu-c committed:
Update proposal text file

susan-shu-c committed:
Update generated files

susan-shu-c commented:
hi, for now, I have marked the `tool.call[...]` fields as OTel related - in v1.37.0 it'd still be under `gen_ai.operation.name` (roughly speaking) - [link](https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/)

<img width="1057" height="857" alt="Screenshot 2025-10-17 at 12 05 13 PM" src="https://github.com/user-attachments/assets/b305c410-ba4b-49ca-aa3c-897ff584bb3b" />

mjwolf committed:
Merge remote-tracking branch 'upstream/main' into additional-gen_ai-stage-2

Mikaayenson commented:
> @Mikaayenson any suggestions for workaround?

Without complicated ESQL queries, we may have to develop custom ingest pipelines to concat the role and content fields (especially with messages being variable length).

There is another fundamental issue where ESQL doesn't currently support type `nested` per the docs https://www.elastic.co/docs/reference/query-languages/esql/limitations#_unsupported_types.

IINM, there are no native ESQL ways to walk the array and keep each message's role paired with the content. 

ESQL does support type `text`, but that is for strings, not arrays of objects, so each message would have to serialized, which throws away the structure we get with type `nested`. Aggregations also become problematic and the type change diverges from otel. 

On a different topic, I also think we need to include other fields (e.g. [bedrock](https://github.com/elastic/integrations/blob/main/packages/aws_bedrock/data_stream/invocation/fields/fields.yml)) or at least [used in our prebuilt rules](https://github.com/search?q=repo%3Aelastic%2Fdetection-rules+gen_ai&type=code).  Examples:
- `gen_ai.guardrail_id`
- `gen_ai.policy.*`
- `gen_ai.compliance.*`

Mikaayenson commented on experimental/generated/beats/fields.ecs.yml:
```diff
@@ -3919,20 +3942,57 @@
       default_field: false
+    - name: system_instructions
+      level: extended
+      type: flattened
```
Note: Some of the points I brought up inhttps://github.com/elastic/ecs/pull/2532#issuecomment-3434285628 will apply to flattened as well.

trisch-me commented:
@AlexanderWert can you get your input regarding the type for complex values such as `gen_ai.input.messages` and others from o11y perspective?

`gen-ai.*` fields will be used probably everywhere in our stack and this is first time we are mapping `any` type from otel to ecs, so I think broader audience is needed to make a proper decision

Mikaayenson reviewed (commented):
After talking with @trisch-me and @joe-desimone, we need to do two additional things:
1. Get input from the ESQL team on their ability to support nested/flattened types
2. Get input from the other solutions (observability and search) to weigh in on their use cases.

AlexanderWert commented on schemas/gen_ai.yml:
```diff
@@ -236,3 +236,136 @@
+      level: extended
+      beta: This field is beta and subject to change.
+      otel:
+        - relation: related
```
same as above: why is the relation `related` instead of `match` ?
https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/#gen-ai-tool-call-arguments

AlexanderWert commented on schemas/gen_ai.yml:
```diff
@@ -236,3 +236,136 @@
+      level: extended
+      beta: This field is beta and subject to change.
+      otel:
+        - relation: related
```
Why `related` and not `match` ?

https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/#gen-ai-tool-call-result

susan-shu-c replied:
Thank you, I have updated the otel relation to match [link to v1.38 release notes containing these fields](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.38.0) for reference

AlexanderWert reviewed (dismissed):
See some comments on the OTel relation of some of the attributes / fields.

Also, please give us a bit more, time! I'd like to discuss this within the Observability / OTel team, since with this OTel `any`-typed attributes it's a new use case for our OTel ingest handling.

I'd like us to make sure we pick a proper type for these attributes so we don't run into issues later.

trisch-me commented on schemas/gen_ai.yml:
```diff
@@ -236,3 +236,136 @@
+      level: extended
+      beta: This field is beta and subject to change.
+      otel:
+        - relation: related
```
It wasn’t released yet when this PR was created, now it can be defined as `match`. The same for comments below

joe-desimone commented:
> @joe-desimone will that dependency on `_source` be negatively impacted if LogsDB indexing mode (specifically synthetic _source) is enabled on the deployment?

Good question @MikePaquette. I have an assumption from an ECS perspective we would be ok since afaik we moved to an opt-in for synthetic source for array fields https://github.com/elastic/ecs/issues/2376. But we should ensure this is consistent across o11y/search as well. @andrewkroh any concerns?

susan-shu-c commented:
Just got back from PTO, thanks a lot for the discussion folks. It seems like there are many considerations and I'll draft up a summary and see what we'll need. Brief points:

- ES|QL doesn't support nested/flattened fields; not very high up on the roadmap at this time
- ES|QL supports calling an LLM with `COMPLETION` which can be used for unique ways of evaluating if prompts are malicious that are more flexible than using pattern matching queries
- ES|QL doesn't support remaining `_source`

From Security team, rule writing use case / perspective, to leverage ES|QL features that other query languages cannot, OTel (and porting OTel to ECS via this PR) doesn't work well out of the box. 
However, we'd like to keep ECS in sync with OTel and not introduce divergence.

For now, I'll gather more information on the requirements from a use case perspective and discuss next steps with all.

susan-shu-c committed:
Merge remote-tracking branch 'origin/main' into additional-gen_ai-stage-2

susan-shu-c committed:
Update to flattened types, update rfc to include new advancements and tradeoffs

susan-shu-c committed:
Run make clean

susan-shu-c commented:
Hi @trisch-me all 6 fields introduced in this PR are in the newest v1.40.0 OTel with the exact same naming [link](https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/#gen-ai-system-instructions)

susan-shu-c committed:
Merge branch 'main' into additional-gen_ai-stage-2

susan-shu-c committed:
Resolve merge conflicts

susan-shu-c committed:
Update proposal stage

susan-shu-c committed:
Merge branch 'additional-gen_ai-stage-2' of github.com:elastic/ecs into additional-gen_ai-stage-2

kgeller reviewed (commented):
Thanks for the thorough investigation and write-up on the flattened vs nested decision. Before this gets to the finish line, a few things on the RFC markdown itself ([`rfcs/text/0052-gen_ai-additional-fields.md`](rfcs/text/0052-gen_ai-additional-fields.md)) that I'd like to see addressed per the [RFC template](https://github.com/elastic/ecs/blob/main/rfcs/0000-rfc-template.md):

### 1. Missing template sections

The `## Summary`, `## Usage`, and `## Scope of impact` sections are currently empty or absent. Could you fill these in? Specifically:

- **Summary** — what is this change and why is it being made? (2–5 sentences)
- **Usage** — how will these fields be used in practice? This is one of the key questions reviewers evaluate per [`rfcs/PROCESS.md`](rfcs/PROCESS.md) ("Does this change provide enough utility for its intended use cases?").
- **Scope of impact** — who and what is affected by these changes? The template asks for 2–5 sentences each on ingestion, usage, and the ECS project itself.

### 2. Justification for `flattened` without leaf fields

Since this PR was opened, [#2594](https://github.com/elastic/ecs/pull/2594) (prompted by [#2595](https://github.com/elastic/ecs/issues/2595)) added a requirement to the RFC template for any `flattened` or `object` field without defined children to justify the choice. The pieces are already here — the Concerns section explains why `nested` is ruled out and the YAML examples show the expected shapes — but could you add a short paragraph in `## Fields` that ties them together so it's clear the requirement was considered?

### 3. Consider expanding the Concerns section

Not a blocker since the PR thread serves as living documentation, but a few other things came up in review that might be worth a line or two in the Concerns section:

- Role ↔ content association loss with `flattened`
- Synthetic `_source` cost under LogsDB

And one question: [elastic/elasticsearch#144245](https://github.com/elastic/elasticsearch/pull/144245) was closed in favor of an alternative approach, does that change anything for this proposal?

susan-shu-c committed:
Tidy up RFC, add info per reviewer

susan-shu-c committed:
Formatting

susan-shu-c committed:
Update on new flattened support ES

susan-shu-c committed:
Merge branch 'main' into additional-gen_ai-stage-2

susan-shu-c commented:
@kgeller I've addressed the feedback in https://github.com/elastic/ecs/pull/2532/#pullrequestreview-4124282460 please let me know if I've missed something specific!

kgeller reviewed (approved):
Thanks for the updates @susan-shu-c ! This PR looks good to me. 

Just an FYI since I think it's changed since your previous RFC PRs, this will still require an additional approval (we require 2 for RFCs / schema changes).

trisch-me commented:
Are we using `gen_ai.system` currently? it’s deprecated in semconv in favor for `provider`. Do we want to follow semconv or we stay with old name?

susan-shu-c commented:
@trisch-me happy to add `gen_ai.provider.name` [link](https://opentelemetry.io/docs/specs/semconv/registry/attributes/gen-ai/#gen-ai-provider-name) [Edit: Added!]

If it's in the scope of this PR, how can I incorporate removing `gen_ai.system` ([link](https://www.elastic.co/docs/reference/ecs/ecs-gen_ai#field-gen-ai-system)), in this PR? [Edit: I've made the replacement, but looking for comments if that's the right way to replace a field.]

susan-shu-c committed:
Rename gen_ai.system to gen_ai.provider.name due to OTel deprecation. Revert if not in scope

susan-shu-c committed:
Update generated build files

trisch-me commented on schemas/gen_ai.yml:
```diff
@@ -179,15 +179,14 @@
       otel:
         - relation: match
-    - name: system
+    - name: provider.name
```
@kgeller we don’t have in ECS a proper depreciation and / or rename, do we?

kgeller replied:
Unfortunately no

trisch-me replied:
what is the process of deprecation then? I mean just renaming and that’s it? Any version constraints etc?

kgeller replied:
this field was/is only beta, where we don't provide any guarantees so I don't think there is a need for anything fancy 

If we really want to, we can add the fields side by side with a deprecation notice in the description like we have done in the past ([example](https://github.com/elastic/ecs/pull/1699/changes#diff-9c440f9668f1c0568f953b8a41f930fcdcbeb5c7b84c87a0ea364ac5e7f54970L7462))

susan-shu-c replied:
@kgeller @trisch-me In that case I will keep this PR  as is, unless anyone has strong preference to do something like the example linked

trisch-me replied:
I’m fine with just renaming

trisch-me commented on rfcs/text/0052-gen_ai-additional-fields.md:
```diff
@@ -1,220 +1,184 @@
+### In-line references for this RFC
+
+[1] https://github.com/elastic/ecs/pull/2532#issuecomment-4121325575
+[2] GH issue can be provided upon request
```
what is meant here?

susan-shu-c replied:
It is an internal Elastic issue, but I think it is confusing here, I will remove mention of it

trisch-me reviewed (approved):
lgtm with some minor comments

susan-shu-c committed:
Remove unlinked GH issue

trisch-me committed:
Merge branch 'main' into additional-gen_ai-stage-2

trisch-me commented:
@AlexanderWert @Mikaayenson any comments from your side?

trisch-me committed:
Merge branch 'main' into additional-gen_ai-stage-2

Mikaayenson commented on docs/reference/ecs-gen_ai.md:
```diff
@@ -36,9 +39,12 @@ This field group definition is based on the Gen AI namespace of the OpenTelemetr
 | $$$field-gen-ai-response-id$$$ [gen_ai.response.id](#field-gen-ai-response-id) | _This field is beta and subject to change._ The unique identifier for the completion.<br><br>type: keyword<br><br>example: `chatcmpl-123`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/match-93c93e?style=flat)](/reference/ ...
 | $$$field-gen-ai-response-model$$$ [gen_ai.response.model](#field-gen-ai-response-model) | _This field is beta and subject to change._ The name of the model that generated the response.<br><br>type: keyword<br><br>example: `gpt-4-0613`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/match-93c93e?style=fl ...
-| $$$field-gen-ai-system$$$ [gen_ai.system](#field-gen-ai-system) | _This field is beta and subject to change._ The Generative AI product as identified by the client or server instrumentation.<br><br>type: keyword<br><br>example: `openai`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![equivalent](https://img.shields.io/badge/equivalent-1ba ...
+| $$$field-gen-ai-system-instructions$$$ [gen_ai.system_instructions](#field-gen-ai-system-instructions) | _This field is beta and subject to change._ The system message or instructions provided to the GenAI model separately from the chat history.<br><br>type: flattened<br><br>example: `{  "type": "text",  "content": "You are an Agent that greet users, always use greetings tool to respond"}`<br>< ...
```
Is there any reason we don't keep a deprecated alias? It may help updates with integrations, endpoint package schema etc.

susan-shu-c replied:
Good point Mika, though there isn't a way to really gracefully deprecate, it was discussed here to rename directly: https://github.com/elastic/ecs/pull/2532#discussion_r3161795136

Mikaayenson commented on docs/reference/ecs-gen_ai.md:
```diff
@@ -36,9 +39,12 @@ This field group definition is based on the Gen AI namespace of the OpenTelemetr
 | $$$field-gen-ai-response-id$$$ [gen_ai.response.id](#field-gen-ai-response-id) | _This field is beta and subject to change._ The unique identifier for the completion.<br><br>type: keyword<br><br>example: `chatcmpl-123`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/match-93c93e?style=flat)](/reference/ ...
 | $$$field-gen-ai-response-model$$$ [gen_ai.response.model](#field-gen-ai-response-model) | _This field is beta and subject to change._ The name of the model that generated the response.<br><br>type: keyword<br><br>example: `gpt-4-0613`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/match-93c93e?style=fl ...
-| $$$field-gen-ai-system$$$ [gen_ai.system](#field-gen-ai-system) | _This field is beta and subject to change._ The Generative AI product as identified by the client or server instrumentation.<br><br>type: keyword<br><br>example: `openai`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![equivalent](https://img.shields.io/badge/equivalent-1ba ...
+| $$$field-gen-ai-system-instructions$$$ [gen_ai.system_instructions](#field-gen-ai-system-instructions) | _This field is beta and subject to change._ The system message or instructions provided to the GenAI model separately from the chat history.<br><br>type: flattened<br><br>example: `{  "type": "text",  "content": "You are an Agent that greet users, always use greetings tool to respond"}`<br>< ...
```
This field in particular reuses system but changes the meaning and type.Maybe we should be more explicit in the PR summary about this.

susan-shu-c replied:
Ah this is just an artifact of the git diff; 
1. Rename: gen_ai.system → gen_ai.provider.name
2. Brand new: gen_ai.system_instructions

The git diff makes it seem like it's gen_ai.system -> gen_ai.system_instructions but that's not the case.

Mikaayenson commented on docs/reference/ecs-gen_ai.md:
```diff
@@ -20,8 +20,11 @@ This field group definition is based on the Gen AI namespace of the OpenTelemetr
 | $$$field-gen-ai-agent-description$$$ [gen_ai.agent.description](#field-gen-ai-agent-description) | _This field is beta and subject to change._ Free-form description of the GenAI agent provided by the application.<br><br>type: keyword<br><br>example: `Helps with math problems; Generates fiction stories`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opente ...
 | $$$field-gen-ai-agent-id$$$ [gen_ai.agent.id](#field-gen-ai-agent-id) | _This field is beta and subject to change._ The unique identifier of the GenAI agent.<br><br>type: keyword<br><br>example: `asst_5j66UpCpwteGg4YSxUnt7lPY`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/match-93c93e?style=flat)](/re ...
 | $$$field-gen-ai-agent-name$$$ [gen_ai.agent.name](#field-gen-ai-agent-name) | _This field is beta and subject to change._ Human-readable name of the GenAI agent provided by the application.<br><br>type: keyword<br><br>example: `Math Tutor; Fiction Writer`<br><br>![OTel Badge](https://img.shields.io/badge/OpenTelemetry-4a5ca6?style=flat&logo=opentelemetry) [![match](https://img.shields.io/badge/ ...
+| $$$field-gen-ai-input-messages$$$ [gen_ai.input.messages](#field-gen-ai-input-messages) | _This field is beta and subject to change._ The chat history provided to the model as an input.<br><br>type: flattened<br><br>example: `{  "role": "user",  "parts": [    {      "type": "text",      "content": "Weather in Paris?"    }  ]}, {  "role": "assistant",  "parts": [    {      "type": "tool_call",   ...
```
nit: Does the data really show up like this `" call_` with the space in it?

susan-shu-c replied:
Good catch I will fix this, it's a typo - will fix in ~~`rfcs/text/0052/gen_ai.yaml`~~ the schema file as this location is automatically generated from that

Mikaayenson reviewed (approved):
In general lgtm. @susan-shu-c can you answer the PR summary questions 4-6 prior to merging?

susan-shu-c committed:
Merge branch 'additional-gen_ai-stage-2' of github.com:elastic/ecs into additional-gen_ai-stage-2

susan-shu-c committed:
Remove space

trisch-me committed:
Merge branch 'main' into additional-gen_ai-stage-2

susan-shu-c committed:
Merge branch 'additional-gen_ai-stage-2' of github.com:elastic/ecs into additional-gen_ai-stage-2

susan-shu-c committed:
Fix typo (extra space) and regenerate files
